d2graph/d2graph.go

```diff
@@ -90,11 +90,12 @@ type Object struct {
 	LabelDimensions d2target.TextDimensions `json:"label_dimensions"`
 	References      []Reference             `json:"references,omitempty"`
 
-	*geo.Box      `json:"box,omitempty"`
-	LabelPosition *string `json:"labelPosition,omitempty"`
-	LabelWidth    *int    `json:"labelWidth,omitempty"`
-	LabelHeight   *int    `json:"labelHeight,omitempty"`
-	IconPosition  *string `json:"iconPosition,omitempty"`
+	*geo.Box              `json:"box,omitempty"`
+	LabelPosition         *string `json:"labelPosition,omitempty"`
+	LabelWidth            *int    `json:"labelWidth,omitempty"`
+	LabelHeight           *int    `json:"labelHeight,omitempty"`
+	IconPosition          *string `json:"iconPosition,omitempty"`
+	IsInsideNearContainer bool    `json:"isInsideNearContainer,omitempty"`
```

Comment:
shouldn't need to have this variable. if necessary, you can define something similar to this method:
https://github.com/terrastruct/d2/blob/718daee97c688af1d22605139ffb4c3d71e27c13/d2graph/seqdiagram.go#L9

Reply:
isn't it more efficient if we store this info? only one time iteration is enough.

Reply:
it is, but it's negligible. better to keep data structure minimal in this case

Reply:
fixed, defined as method  `IsInsideNearContainer ` instead